eva-stats/stats_variant_warehouse.py

```diff
@@ -0,0 +1,181 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import psycopg2
+import click
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query, execute_query
+from ebi_eva_common_pyutils.mongo_utils import get_mongo_connection_handle
+from ebi_eva_common_pyutils.config_utils import get_pg_metadata_uri_for_eva_profile, get_properties_from_xml_file
+from ebi_eva_common_pyutils.logger import logging_config
+
+
+logger = logging_config.get_logger(__name__)
+
+
+def get_database_name_from_assembly(metadata_handle, assembly):
+    query = "select t.taxonomy_code, a.assembly_code " \
+            "from assembly a " \
+            "left join taxonomy t on (t.taxonomy_id = a.taxonomy_id)" \
+            "where assembly_accession = '{0}'".format(assembly)
+    result = get_all_results_for_query(metadata_handle, query)
+    database_name = 'eva_{0}_{1}'.format(result[0][0], result[0][1])
+    return database_name
+
+
+def get_mongo_primary_host_and_port(mongo_hosts_and_ports):
+    """
+    :param mongo_hosts_and_ports: All host and ports stored in the private settings xml
+    :return: mongo primary host and port
+    """
+    for host_and_port in mongo_hosts_and_ports.split(','):
+        if '001' in host_and_port:
+            properties = host_and_port.split(':')
+            return properties[0], properties[1]
+
+
+def get_from_variant_warehouse(mongo_handle, metadata_handle, projects):
+    projects_evapro = "('" + projects.replace(",", "','") + "')"
+    query = "select a.vcf_reference_accession, p.project_accession, a.analysis_accession, af.file_id, " \
+            "f.filename, f.file_type, asm.taxonomy_id " \
+            "from analysis a " \
+            "left join analysis_file af on (af.analysis_accession = a.analysis_accession)" \
+            "left join file f on (f.file_id = af.file_id)" \
+            "left join project_analysis p on (p.analysis_accession = a.analysis_accession)" \
+            "left join assembly asm on (asm.assembly_accession = a.vcf_reference_accession)" \
+            "where f.file_type = 'VCF' " \
+            "and p.project_accession in {0}" \
+            "order by a.vcf_reference_accession, p.project_accession, a.analysis_accession".format(projects_evapro)
+    logger.info("Querying EVAPRO")
+    result_evapro = get_all_results_for_query(metadata_handle, query)
+
+    analyses = []
+    analysis_filename = {}
+    for index, row in enumerate(result_evapro):
```

Comment:
For this I think you want to use the `itertools.groupby` function.
It assumes that list is sorted by the value you're grouping on but it will simplify your code.
```
from itertools import groupby 
from operator import itemgetter

for project, rows_for_project in groupby(result_evapro, key=itemgetter(1)):
    analyses = [row[2] for row in rows_for_project]
    filenames = set([row[4] for row in rows_for_project])
```

Reply:
Thanks, changed it to use `itertools.groupby` and it did simplified the code